pcp/PCPDynamicMeter.c

```diff
@@ -458,3 +476,25 @@ void PCPDynamicMeter_display(PCPDynamicMeter* this, ATTR_UNUSED const Meter* met
    if (nodata)
       RichString_writeAscii(out, CRT_colors[METER_VALUE_ERROR], "no data");
 }
+
+char** PCPDynamicMeter_getChoices(PCPDynamicMeter* this) {
+   unsigned int indom = this->metrics->indom;
+   int count = 0;
+   int* in;
+   char** raw;
+
+   count = pmGetInDom(indom, &in, &raw);
+   if (count < 0)
+      return NULL;
+
+   char** ret = xMalloc(count*sizeof(char*));
+
+   for (int i = 0; i < count; i++)
+      ret[i] = xStrdup(raw[i]);
+   ret[count] = NULL;
```

Comment:
I tried to return raw directly, but I got a Segfault. I don't know why! 
So I had to copy `raw` into `ret`

Reply:
Probably because `pmGetInDom` sets up references in `raw` to point into the same memory allocation.